Xamarin.Forms.Platform.Android/Renderers/ImageRenderer.cs

```diff
@@ -71,40 +71,52 @@ void UpdateAspect()
 			Bitmap bitmap = null;
 
 			ImageSource source = Element.Source;
-			IImageSourceHandler handler;
 
 			if (previous != null && Equals(previous.Source, Element.Source))
 				return;
 
 			((IImageController)Element).SetIsLoading(true);
 
 			var formsImageView = Control as FormsImageView;
-			if (formsImageView != null)
-				formsImageView.SkipInvalidate();
+			formsImageView?.SkipInvalidate();
 
 			Control.SetImageResource(global::Android.Resource.Color.Transparent);
 
-			if (source != null && (handler = Registrar.Registered.GetHandler<IImageSourceHandler>(source.GetType())) != null)
+			if (source != null)
 			{
-				try
+				IImageSourceHandler handler;
+				if ((handler = Registrar.Registered.GetHandler<IImageSourceHandler>(source.GetType())) != null)
 				{
-					bitmap = await handler.LoadImageAsync(source, Context);
+					try
+					{
+						bitmap = await handler.LoadImageAsync(source, Context);
+					}
+					catch (TaskCanceledException)
+					{
+					}
+					catch (IOException ex)
+					{
+						Log.Warning("Xamarin.Forms.Platform.Android.ImageRenderer", "Error updating bitmap: {0}", ex);
+					}
 				}
-				catch (TaskCanceledException)
-				{
-				}
-				catch (IOException ex)
+
+				if (bitmap == null && source is FileImageSource)
 				{
-					Log.Warning("Xamarin.Forms.Platform.Android.ImageRenderer", "Error updating bitmap: {0}", ex);
+					Control.SetImageResource(ResourceManager.GetDrawableByName(((FileImageSource)Element.Source).File));
```

Comment:
The problem with that approach is that converting vectors to bitmaps is an expensive procedure especially on the `argb8888` setting which allocates 4 bytes per pixel. If you had a list of vectors and you navigated to and from the page frequently, it's easy to end up with an `OutOfMemoryError` exception. Of course, one solution would be to cache the results. `SetImageResource()` seems to be doing everything including resizing out of the box. However, the downside is that decoding is done on the UI thread. If this creates a latency concern, then we'd need to consider caching one again.

This change is by no means an advanced solution. It's just providing basic support for vectors.